tools/topology/topology2/platform/intel/dmic-generic.conf

```diff
@@ -198,16 +211,112 @@ IncludeByKey.PASSTHROUGH {
 					type		dai_out
 					stream_name	$DMIC0_NAME
 					node_type $DMIC_LINK_INPUT_CLASS
+					num_input_audio_formats	2
+					Object.Base.input_audio_format [
+						{
+							in_rate			$DMIC0_RATE
+							in_bit_depth		32
+							in_valid_bit_depth	32
+						}
+						{
+							in_rate			$DMIC0_RATE
+							in_channels		4
+							in_bit_depth		32
+							in_valid_bit_depth	32
+							in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+							in_ch_map	$CHANNEL_MAP_3_POINT_1
+						}
+					]
+					num_output_audio_formats	2
+					Object.Base.output_audio_format [
+						{
+							out_rate		$DMIC0_RATE
+							out_bit_depth		32
+							out_valid_bit_depth	32
+						}
+						{
+							out_rate		$DMIC0_RATE
+							out_channels		4
+							out_bit_depth		32
+							out_valid_bit_depth	32
+							out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+							out_ch_map	$CHANNEL_MAP_3_POINT_1
+						}
+					]
```

Comment:
this part adds a lot more than the simple addition of a non-default 48kHz.

That should be moved in a separate patch with explanations on why we didn't need this earlier?

Reply:
Yep, splitting this to two.